BoardStateClass.py

```diff
@@ -0,0 +1,179 @@
+import numpy as np
+from ColumnClass import Column
+from SlotClass import Slot
+#from CheckForWin import check_line_for_win
+
+
+class Board_State:
+
+    _row: str
+    _base: str
+    _game_ended: bool
+    _spaces_filled: int
+    _max_spaces_filled: int
+    _game_board: Slot
+
+    def __init__(self, board_width, board_height, win_length):
+        self._board_width = board_width
+        self._board_height = board_height
+        self._win_length = win_length
+        self._spaces_filled = 0
+        self._max_spaces_filled = board_width * board_height
+
+        self._cols = []
+        self._create_columns()
+
+    def _create_columns(self):
+        for i in range(0, self._board_width):
+            _col = Column(i+1, self._board_height, self._win_length)
+            self._cols.append(_col)
+
+    def create_board_array(self):
+        '''
+        Returns an array of slots representing the game board. Information taken from the columns.
+        :return:
+        '''
+        game_board = np.array(Slot)
+        temp_list_board = []
+        for col in self._cols:
+            col = np.array(col.get_slots())
+            temp_list_board.append(col)
+        game_board = np.transpose(np.array(temp_list_board))
+        return game_board
+
+    def insert_counter(self, column_no, player):  #type --> bool
+        success = False
+        if(1 <= column_no <= self._board_width):
+            #Player selected valid column
+            chosen_column = self._cols[column_no-1]  # Column labels 1-indexed whereas array is 0-indexed
+            if not chosen_column.is_full():
+                #Chosen column has space.
+                chosen_column.add_counter(player)
+                self._spaces_filled += 1
+                success = True
+            else:
+                print("That column is already full, choose another!")
+        else:
+            print("Invalid selection, please choose another column")
+        return success
+
+    def game_drawn(self):
+        if self._spaces_filled >= self._max_spaces_filled:
+            print("It\'s a draw.")
+            self._game_ended = True
+            return True
+        else:
+            return False
+
+    def game_won(self):
+        won = False
+        #Collect all possible lines in which a win could occur
+        lines = self._get_rows() + self._get_columns() + self._get_diagonals()
```

Comment:
consider having a get lines method. you can call the columns, rows etc in that. should probably be encapsulated/moved into a "Model" (MVC) that describes the current state without the Business logic (win conditions etc)